### ISSUE TITLE
Addon structure

### ISSUE THREAD

**Issue by tokorokot** [Resike/PVPSound#9]
It can be difficult to fix BGs, because pieces of code for each BG are scattered over 4000 lines length file. What do you think about taking out BGs code in separated lua files?
And it's also possible to move 'registerevent' funcs to these files to register event only when it's needed, and not on addon load.

**Comment by Resike** [comment]
Yeah i wanted to rewrite the addon since ages, but I personally don't have the time for it right now.

**Comment by tokorokot** [comment]
I'll try to make something with it in my fork, but I don' t have much time at the moment too, so it seems to be not very fast process:)

**Comment by tokorokot** [comment]
Have some progress in work with new structure.  Pushed it to new branch in my fork.
